pkg/informer/informer.go

```diff
@@ -194,6 +176,32 @@ func NewDynamicDiscoverySharedInformerFactory(
 	return f, nil
 }
 
+func byGroupFirstFoundVersionResourceIndexFunc(obj interface{}) ([]string, error) {
+	crd, ok := obj.(*apiextensionsv1.CustomResourceDefinition)
+	if !ok {
+		return nil, fmt.Errorf("%T is not a CustomResourceDefinition", obj)
+	}
+
+	firstServedVersion := ""
+	for _, version := range crd.Spec.Versions {
+		if !version.Served {
+			continue
+		}
+		firstServedVersion = version.Name
+		break
+	}
+
+	if firstServedVersion == "" {
+		return []string{}, nil
+	}
+
+	group := crd.Spec.Group
+	resource := crd.Spec.Names.Plural
+
+	indexValue := fmt.Sprintf("%s/%s/%s", group, firstServedVersion, resource)
```

Comment:
We've found a lot of benefit to breaking out these index formatting functions so that users who want to query the index can ensure they're doing the right thing.

Reply:
Done 👍🏼.